README-dev.md

```diff
@@ -66,57 +91,46 @@ The wrappers are built with the plugin. `dvp build` has a hidden `--dev` flag. T
+1. Install the package's development dependencies: `pip install -r requirements.txt`.
+2. Install the package itself in editable mode: `pip install -e .`.
+3. Run unit tests: `python -m pytest src/main/python`.
 
-At the moment blackbox refers to a property file in the app-gate to determine the version of the SDK to install for tests so this property always needs to updated for automated testing.
+Currently, there's no way to locally run unit tests in all packages with one command. However, soon, there will be a Github Action which will run all pre-commit checks on a pull request. 
 
-NOTE: The app-gate does not pull in the wrappers or CLI from this repository.
+### Testing sdk-gate changes with app-gate code
 
-The easiest way to do both of these is:
-
-1. Update the version of the SDK to something unique and clearly a development build. The standard is `x.y.z-internal-abc-<your_name>`. For example, `1.1.0-internal-001-grant`.
-2. Run `./gradlew publishDebug` from the root of this repository.
-3. In `appliance/gradle.properties` in the app-gate update `virtualizationSdkVer` to match the SDK version.
-
-Run an appliance-update for manual testing and/or kick off automated blackbox tests by running `git blackbox -s appdata_python_samples` from your app-gate development branch.
-
-
-## SDK Review and Provisional app-gate review
-
-Once you're finished with local development and testing, you can publish your final SDK review to reviewboard.
-
-In addition, it's customary to publish a "provisional" appgate review, so that people can get insight into how the out-for-review SDK changes will actually be used by the appgate. Of course, this review will contain all your temporary local-build changes mentioned above. So, in your review, you'll want to mention that these temporary changes will be reverted before the review is finalized.
-
-## Pushing and Deploying SDK Code
+#### Manual testing
 
+Run `dvp build --dev` to build your plugin and then upload it to a Delphix Engine to test.
 
-### Publishing
+The wrappers are built with the plugin. `dvp build` has a hidden `--dev` flag. This builds `common`, `libs`, and `platform` locally and bundles them with the plugin. A special configuration entry is needed in your dvp config file which is located at `~/.dvp/config`:
 
-There are two Gradle tasks that do publishing: `publishDebug` and `publishProd`. They differ in two ways:
+```
+[dev]
+vsdk_root = /path/to/vsdk_repo_root
+```
 
-1. They publish the Python distributions to separate repositories on Artifactory. `publishDebug` uploads to `dvp-local-pypi`. This is a special repository that has been setup to test the SDK. It falls back our our production PyPI repository, but artifacts uploaded to `dvp-local-pypi` do not impact production artifacts. This should be used for testing. `publishProd` does upload the Python distributions to our production Artifactory PyPI repository, `delphix-local`.
+##### Functional (blackbox) testing
```

Comment:
Similar to the above question: Is this the best way to run blackbox tests?

Reply:
Ah is this because blackbox still relies on the gradle build?

Reply:
That's correct. I had initially described the process you suggested - building from a branch on GitHub - but that won't work until balckbox can utilize the new Python build.